test/libsolidity/SemanticTest.h

```diff
@@ -70,6 +87,10 @@ class SemanticTest: public SolidityExecutionFramework, public EVMVersionRestrict
 	bool m_runWithABIEncoderV1Only = false;
 	bool m_allowNonExistingFunctions = false;
 	bool m_compileViaYulCanBeSet = false;
+
+	bool m_gasCostFailure = false;
+	bool m_enforceGasCost = false;
```

Comment:
I see that @ekpyron suggested adding it in https://github.com/ethereum/solidity/pull/10863/files#r573604188

I'm not really sure why. Constructor already takes care of that and I think that having defaults in too many places just makes it easy for them to go out of sync.

Reply:
It's **always** better to immediately member-initialize PODs, everywhere, no matter what, no exceptions whatsoever.
If you do, the worst thing that can happen is that you overwrite it by the opposite value in the constructor, but there's really no harm in that. The worst thing that can happen in *not* doing it, is undefined behaviour that will not break for ten years and then all of a sudden make your program explode for no reason ;-).

Reply:
Actually the worst that it can happen is that you can forget to actually initialize it with the proper value in constructor and not notice because it won't fail. But yeah, I get your point, in C++ uninitialized stuff will likely not fail outright anyway and might cause trouble anyway. I guess zero-initialization might be safer as a general rule. The biggest issue I had with this was the repeated default value for minimum gas cost but that's gone after the previous review round.

Reply:
I would say, you can forget to initialize a value in a constructor no matter if you member-initialized it or not, that's the same risk independently. Only in one case what you end up with fails deterministically, in the order randomly :-).
Or put yet in another way: *if* you forget to initialize something in the constructor, but *haven't* member-initialized, your bug is as bad as choosing any possible value for it or the one worst possible value that might really break stuff, so you're up against an entire array of potential misbehaviour - whereas if you do member-initialize, you see and know exactly what will happen and can check how bad it is easily.
So yeah, as you see, I'm an (exaggeratedly ;-)) strong proponent of this ;-).